cmd/sonobuoy/app/gen.go

```diff
@@ -133,6 +136,25 @@ func (g *genFlags) Config() (*client.GenConfig, error) {
 	}, nil
 }
 
+// resolveConformanceImage maps versions before 1.13 to Heptio's image and otherwise
+// to the upstream cnoformance image. Latest is always mapped to the upstream
+// regardless. The comparison is just lexical, e.g. "foo" < "v1.13" and "zip" >
+// "v1.13". These are a-typical and not given more support at this time.
+func resolveConformanceImage(imageVersion string) string {
+	// TODO(johnschnake): This logic should be temporary and is only
+	// required as we phase in the use of the upstream k8s kube-conformance
+	// image instead of our own heptio/kube-conformance one. They started
+	// publishing it for v1.13.
+	switch {
+	case imageVersion == ConformanceImageVersionLatest:
```

Comment:
I think we should move to versioned image tags. But if not, could simply this to:

If < 1.13, then use DefaultKubeConformanceImageURL else use UpstreamKubeConformanceImageURL

Reply:
The only problem is if they specify they want to use `latest`; currently they can provide just a version override (not image+version).

Since this is just a string we have to play with here and . a lexical comparison, `"latest" < "v1.13"` and they would get the heptio image then.

Reply:
Not that latest is a great idea, but it is a string which is common for users to specify for an image tag so we should handle that appropriately.

Reply:
Yeah but are you sure `latest` is the latest release? My guess is that latest probably points to something in 1.14 or 1.15 already which is invalid for a 1.13 cluster (even though it might work).

Reply:
Totally agree; again, its not that I think _we_ should be passing latest, but if someone overrides and says `latest` (which seems very possible) I'd rather give them the upstream image than the heptio one.

Reply:
If someone says they want latest that's fine, but sonobuoy should not default to that. For example, if I run a `sonobuoy gen` this is what the plugin gives me, the conformance image is: `gcr.io/google-containers/conformance:latest` which is what version?
```
apiVersion: v1
data:
  e2e.yaml: |
    sonobuoy-config:
      driver: Job
      plugin-name: e2e
      result-type: e2e
    spec:
      env:
      - name: E2E_FOCUS
        value: '\[Conformance\]'
      - name: E2E_SKIP
        value: 'Alpha|\[(Disruptive|Feature:[^\]]+|Flaky)\]'
      - name: E2E_PARALLEL
        value: '1'
      command: ["/run_e2e.sh"]
      **image: gcr.io/google-containers/conformance:latest**
      imagePullPolicy: Always
      name: e2e
      volumeMounts:
      - mountPath: /tmp/results
        name: results
        readOnly: false
      tolerations:
        - operator: "Exists"
```

Reply:
100%, I guess we're having 2 different conversations. I thought this note was specifically about how to handle the switch/if statement.

There is another note above about the use of `latest` as a default value which I support shifting.

I think we're in agreement, right?